install/page_modcheck.php

```diff
@@ -26,135 +26,151 @@
+}
+
+function phpVersionCheck() {
+    if (version_compare( phpversion(), '5.7', '>=')) {
+        return 1;
+    } elseif (version_compare( phpversion(), '5.6', '>=')) {
+        return 0;
+    } else {
+        return -1;
+    }
 }
 
+function successImg($success_val) {
+    $img_name = $success_val >= 0 ? "yes" : "no";
+    return '<img src="img/' . $img_name . '.png" alt="Success?" class="rootimg" />';
+}
+
+$php_version_success = phpVersionCheck();
+$mysql_success = function_exists('mysql_connect') ? 1 : -1;
+$extension_success = extension_loaded('session') ? 1 : -1;
+$pcre_success = extension_loaded('pcre') ? 1 : -1;
+
+$php_version_success_str = xoDiag($php_version_success, phpversion()) . successImg($php_version_success);
+$mysql_success_str = xoDiag($mysql_success) . successImg($mysql_success);
+$extension_success_str = xoDiag($extension_success) . successImg($extension_success);
+$pcre_success_str = xoDiag($pcre_success) . successImg($pcre_success);
+
 ob_start();
 ?>
-<fieldset>
-<h3><?php echo REQUIREMENTS; ?></h3>
-<h4><?php echo SERVER_API; ?>:&nbsp; <?php echo php_sapi_name(); ?> <img
-	src="img/yes.png" alt="Success" class="rootimg" /></h4>
-<div class="clear">&nbsp;</div>
-<h4><?php echo _PHP_VERSION; ?>:&nbsp; <?php
-if (version_compare( phpversion(), '5.2', '>=')) {
-	echo xoDiag( 1, phpversion() );
-} elseif (version_compare( phpversion(), '5.1', '>=')) {
-	echo xoDiag( 0, phpversion() );
-} else {
-	echo xoDiag( -1, phpversion() );
-}
-?> <img
-	src="img/<?php echo (isset($php_version_error) ? "no" : "yes") ?>.png"
-	alt="Success" class="rootimg" /></h4>
-<div class="clear">&nbsp;</div>
-<h4><?php printf( PHP_EXTENSION, 'MySQL' ); ?>:&nbsp; <?php echo xoDiag( function_exists( 'mysql_connect' ) ? 1 : -1 ); ?>
-<img src="img/yes.png" alt="Success" class="rootimg" /></h4>
-<div class="clear">&nbsp;</div>
-<h4><?php printf( PHP_EXTENSION, 'Session' ); ?>:&nbsp; <?php echo xoDiag( extension_loaded( 'session' ) ? 1 : -1 ); ?>
-<img src="img/yes.png" alt="Success" class="rootimg" /></h4>
-<div class="clear">&nbsp;</div>
-<h4><?php printf( PHP_EXTENSION, 'PCRE' ); ?>:&nbsp; <?php echo xoDiag( extension_loaded( 'pcre' ) ? 1 : -1 ); ?>
-<img src="img/yes.png" alt="Success" class="rootimg" /></h4>
-<div class="clear">&nbsp;</div>
-<h4>file_uploads:&nbsp; <?php echo xoDiagBoolSetting( 'file_uploads', true ); ?>
-<img src="img/yes.png" alt="Success" class="rootimg" /></h4>
-<div class="clear">&nbsp;</div>
-</fieldset>
 
-<fieldset>
-<h3><?php echo RECOMMENDED_EXTENSIONS; ?></h3>
-<p><?php echo RECOMMENDED_EXTENSIONS_MSG; ?></p>
-<div class="clear">&nbsp;</div>
+    <fieldset>
+        <h3><?php echo REQUIREMENTS; ?></h3>
+        <h4><?php echo SERVER_API; ?>:&nbsp; <?php echo php_sapi_name(); ?> <img
+                    src="img/yes.png" alt="Success" class="rootimg" /></h4>
+        <div class="clear">&nbsp;</div>
+        <h4><?php echo _PHP_VERSION; ?>:&nbsp;
```

Comment:
Isn't the constant PHP_VERSION? no underscore before it?

Reply:
This was what was in the file originally — maybe a previously committed typo?